source/iopipe/json/serialize.d

```diff
@@ -31,6 +31,95 @@ import std.typecons : Nullable;
 import std.conv;
 import std.format;
 
+struct DefaultDeserializationPolicy {
+    ReleasePolicy relPol = ReleasePolicy.afterMembers; // default policy
+        
+    // Called at beginning of struct/class deserialization
+    bool[SerializableMembers!T.length] onBegin(JT, T)(ref JT tokenizer) {
+
+        // Pre-mark optional fields as visited
+        alias members = SerializableMembers!T;
+        bool[members.length] visited;
+
+        static foreach(idx, m; members) {
+            static if(hasUDA!(__traits(getMember, T, m), optional))
+                visited[idx] = true;
+        }
+        
+        tokenizer.nextSignificant
+            .jsonExpect(JSONToken.ObjectStart, "Parsing " ~ T.stringof);
```

Comment:
Just noticed that this is in the policy callback, and not done in the deserialize function.

This really belongs in the `deserializeImplWithPolicy` function, and not the policy function itself (all other parsing validation is done there). We'll get to this in a later PR, but for now, can this (and the complementary check for the end object token) be moved into the main function?